### ISSUE TITLE
Visual Studio Publish command creates corrupt web.config when publishing to an Azure VM

### ISSUE THREAD

**Issue by masonwheeler** [aspnet/Blazor#612]
Repro steps:

1.  Set up an Azure VM running Windows (not an Azure web app instance--this might work as well, but this is the way I know will work)
2. Configure IIS on the VM, including installing the rewrite module, and follow all other steps necessary to prepare it to receive web updates via Visual Studio's Publish command.  Also install the .Net Core Preview version Blazor needs.
3. Create a Blazor project in Visual Studio.
4. Use Visual Studio's Publish command to publish the Blazor project to your Azure VM
5. Use your browser to attempt to access the Blazor project from your dev PC.

Expected:  The project loads in your browser
Observed: IIS 500 error page

Publish is screwing up here, by publishing a web.config (or is it config.web?  I'm not at the relevant computer at the moment) file that tries to treat the Blazor project as an ASP.Net server project, rather than as static content that's intended to run in the browser.  (After deleting the relevant lines from the config file, I was able to get it working.)

**Comment by SteveSandersonMS** [comment]
> After deleting the relevant lines from the config file, I was able to get it working

Which lines did you delete?

**Comment by masonwheeler** [comment]
@SteveSandersonMS This is a difficult question to answer, as it's been deleted and so is no longer available for me to reference.  It happened a few days ago; I'm just now getting around to writing up the bug report for it.

I think there was a `<handler>` tag that told it how to handle a project of type `aspNetCore`, and then an `<aspNetCore>` tag that told it that it's supposed to be serving my Blazor project DLL as an ASP.Net Core project.  That's the general gist of it, at least; the details might not be exactly right.

**Comment by SteveSandersonMS** [comment]
The web config file published for standalone apps is this: https://github.com/aspnet/Blazor/blob/dev/src/Microsoft.AspNetCore.Blazor.Build/targets/Standalone.Web.config It doesn't contain any `handler`, so I'm not sure how to interpret this report.

Are you able to reproduce the situation and determine what resolution you thought was appropriate?

**Comment by masonwheeler** [comment]
Yeah, that's definitely what the top of my web.config looked like, but after all that stuff was the part I described.

I'm currently not able to reproduce the issue, as something went wrong with my Azure system and it's been shut down since last night, and I can't get ahold of Azure support to get it worked out! 😠

**Comment by SteveSandersonMS** [comment]
OK, please let us know if you do get repro details at some point. Closing until then.

**Comment by xclud** [comment]
The same problem here. Once deployed to my Windows Server 2016, the Standalone app causes a 500 error, while Hosted app works nicely.

**Comment by jcmontx** [comment]
I tried deploying the template to an app service and I'm getting 502.5 error. You can check it here: http://blazordemo1.azurewebsites.net/

**Comment by masonwheeler** [comment]
@xclud @jcmontx Can either of you check the `web.config` file on your ASP.Net server and see if it matches what I posted above, and post it here?  I wasn't able to since I didn't save the old version, so having a file that demonstrates the problem would probably help with tracking it down.

**Comment by xclud** [comment]
@masonwheeler yes i get the same web.config except that instead of `[ServeSubdirectory]` i get `Dashboard\dist\` where Dashboard is my project name.

**Comment by xclud** [comment]
@jcmontx i usually fix 502.5 errors by installing exactly the same .NET Core runtime of my dependencies and giving Everyone permission to my wwwroot folder.

**Comment by xclud** [comment]
@masonwheeler my IIS error code is 500.19 which according to the documentations it is because of non-valid XML in web.config. I double checked the file with an xml-validator and it's just fine. I guess the elements are not recognized by IIS.

**Comment by masonwheeler** [comment]
@xclud Yeah, can you post the actual text of the config file on here, suitably anonymized if necessary, so @SteveSandersonMS will have something to go by?

**Comment by xclud** [comment]
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <staticContent>
      <remove fileExtension=".dll" />
      <remove fileExtension=".json" />
      <remove fileExtension=".wasm" />
      <remove fileExtension=".woff" />
      <remove fileExtension=".woff2" />
      <mimeMap fileExtension=".dll" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".json" mimeType="application/json" />
      <mimeMap fileExtension=".wasm" mimeType="application/wasm" />
      <mimeMap fileExtension=".woff" mimeType="application/font-woff" />
      <mimeMap fileExtension=".woff2" mimeType="application/font-woff" /> 
    </staticContent>
    <httpCompression>
      <dynamicTypes>
        <add mimeType="application/octet-stream" enabled="true" />
        <add mimeType="application/wasm" enabled="true" />
      </dynamicTypes>
    </httpCompression>
    <rewrite>
      <rules>
        <rule name="Serve subdir">
          <match url=".*" />
          <action type="Rewrite" url="Dashboard\dist\{R:0}" />
        </rule>
        <rule name="SPA fallback routing" stopProcessing="true">
          <match url=".*" />
          <conditions logicalGrouping="MatchAll">
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
          </conditions>
          <action type="Rewrite" url="Dashboard\dist\" />
        </rule>
      </rules>
    </rewrite>
  </system.webServer>
</configuration>
```

**Comment by xclud** [comment]
I found the solution. The problem was that my IIS' Rewrite module was gone. It's kind of weird as i am sure i installed the module during IIS setup.

This seems to be a known bug for Windows 10 IIS according to this [Stackoverflow Post](https://stackoverflow.com/questions/35319942/url-rewrite-does-not-install-on-windows-10-iis-10) and this [IIS Forum thread](https://forums.iis.net/t/1223556.aspx). However, my machine does not have Windows 10 installed. I am using Windows Server 2016 latest update.

<img width="393" alt="IIS Config" src="https://user-images.githubusercontent.com/12394889/39672073-aa726546-5138-11e8-82e3-fa6861fbfcbe.PNG">

Enabling IIS Rewrite module as seem above (And making sure it is installed this time) **did not fix my problem** though. I had to install [this module](https://www.microsoft.com/en-us/download/details.aspx?id=47337) to finally get it working.

In addition, I enabled one more module during my try and errors. Dynamic Content Compression seemed to be necessary (to me) as `web.config` uses `<httpCompression><dynamicTypes>...</dynamicTypes></httpCompression>`.  I am not sure if enabling this module has any direct impact here on this problem.
